runtime/src/identity.rs

```diff
@@ -793,22 +748,25 @@ decl_module! {
                                     T::AcceptTransferTarget::accept_ticker_transfer(did, auth_id),
                                 AuthorizationData::TransferTokenOwnership(_) =>
                                     T::AcceptTransferTarget::accept_token_ownership_transfer(did, auth_id),
+                                AuthorizationData::AddMultisigSigner =>
+                                    T::AddSignerMultisigTarget::accept_multi_sig_signer(Signer::from(did), auth_id),
                                 _ => Err("Unknown authorization")
                             };
                         }
                     }
                 },
-                Signer::Key(_key) => {
+                Signer::Key(key) => {
                     for auth_id in auth_ids {
                         // NB: Even if an auth is invalid (due to any reason), this batch function does NOT return an error.
                         // It will just skip that particular authorization.
                         if <Authorizations<T>>::exists((signer, auth_id)) {
-                            let _auth = Self::authorizations((signer, auth_id));
-                            // NB: Result is not handled, invalid auths are just ignored to let the batch function continue.
-                            // TODO: Add cases where signing key is authorized
-                            // let _result = match auth.authorization_data {
-                            //     _ => Err("Unknown authorization")
-                            // };
+                            let auth = Self::authorizations((signer, auth_id));
+                            //NB: Result is not handled, invalid auths are just ignored to let the batch function continue.
+                            let _result = match auth.authorization_data {
+                                AuthorizationData::AddMultisigSigner =>
+                                T::AddSignerMultisigTarget::accept_multi_sig_signer(Signer::from(key), auth_id),
+                                _ => Err("Unknown authorization")
```

Comment:
If this is discarded, what is the goal of computing this error?

Reply:
It basically keeps the code consistent between the cases where we handle errors and where we don't. Also, the rust optimizer is probably just going to optimize it away. I do need to return something in the catch-all case to please the compiler so I might as well just return the proper error.

Reply:
I'd just do that with an `if let AuthorizationData::AddMultisigSigner = auth.authorization_data { ... }`, discard the result of `accept_multi_sig_signer` and remove the `Err(...)` as redundant.

Reply:
That works for this specific case but I'm trying to maintain the same format here. If/when we add more functions here, the match statement will come handy. I don't see any real disadvantage of using match here. The error is not being computed, A fixed error it just being assigned. Even if the compiler doesn't remove it via optimizations, it's a no-op with very little resource usage. I think it's worth keeping it this way for added clarity and explicitness.